packages/components/clipboard/src/components/osds-clipboard/core/controller.ts

```diff
@@ -1,8 +1,50 @@
+import type { OsdsClipboard } from '../osds-clipboard';
 import { writeOnClipboard } from '@ovhcloud/ods-common-core';
 
 class OdsClipboardController {
+  private component: OsdsClipboard;
+
+  constructor(component: OsdsClipboard) {
+    this.component = component
+  }
+
   async handlerClick(value: string): Promise<void> {
-    await writeOnClipboard(value);
+    const successMessage = this.component.el.querySelector('[slot=success-message]')?.innerHTML;
+    const errorMessage = this.component.el.querySelector('[slot=error-message]')?.innerHTML;
+
+    try {
+      writeOnClipboard(value);
+      this.component.surfaceMessage = successMessage;
+      if (this.component.surface && this.component.surfaceMessage !== "") {
+        this.component.surface.opened = !this.component.surface.opened;
+      }
+    } catch (error) {
+      this.component.surfaceMessage = errorMessage;
+      if (this.component.surface  && this.component.surfaceMessage !== "") {
+        this.component.surface.opened = !this.component.surface.opened;
+      }
+      return Promise.reject(error);
```

Comment:
```suggestion
      return throw error;
```

Reply:
Not sure if the return is need with a throw

Reply:
Fixed